### PR TITLE
always upload e2e logs

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
we should always upload e2e logs, whether the e2e is successful, or is failed, or is canceled due to a timeout
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -171,6 +171,7 @@ jobs:
       - run: make e2e
 
       - uses: actions/upload-artifact@v3
+        if: always()
```
Comment (fisherxu) [inline .github/workflows/main.yaml:174]: It will not upload by default if running successed?

Comment (gy95) [reply .github/workflows/main.yaml:174]: no, it will upload logs always,  regardless of success, failure, or cancellation.

Comment (gy95) [reply .github/workflows/main.yaml:174]: without this `if: always()` flag, will only upload logs when the last step is successful

Comment (fisherxu) [reply .github/workflows/main.yaml:174]: Oh, so if without `if: always()`, when `make e2e` run failed, this upload-artifact wil be skipped?

Comment (gy95) [reply .github/workflows/main.yaml:174]: yes
ref: https://github.com/kubeedge/kubeedge/actions/runs/2501087906

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**